lib/replica.rb

```diff
@@ -207,11 +211,10 @@ def compile_revision_ids_query(revisions)
   # These are typical network errors that we expect to encounter.
   TYPICAL_ERRORS = [Errno::ETIMEDOUT, Net::ReadTimeout, Errno::ECONNREFUSED,
                     Oj::ParseError].freeze
-  def report_exception(error, endpoint, query, level='error')
-    level = 'warning' if TYPICAL_ERRORS.include?(error.class)
-    Raven.capture_exception error, level: level, extra: {
-      query: query, endpoint: endpoint, language: @wiki.language, project: @wiki.project
-    }
-    return nil
+
+  def invoke_error_handling(error, endpoint, query: nil, url: nil, response_body: nil)
+    extra = { query: query, endpoint: endpoint, language: @wiki.language, project: @wiki.project }
+    optional_params = build_optional_params(@course, error, url, response_body)
+    perform_error_handling(error, TYPICAL_ERRORS, extra, @course, optional_params)
```

Comment:
This would definitely be nicer if the module could find TYPICAL_ERRORS for whichever class it has been included in, automatically, instead of need it as an explicit argument.

Reply:
Done!